src/metaswitch/clearwater/etcd_shared/plugin_utils.py

```diff
@@ -25,12 +25,11 @@ def run_command(command, namespace=None, log_error=True):
     passing to this function.
     """
     if namespace:
-        command = "ip netns exec {} ".format(namespace) + command
+        command = "ip netns exec {} ".format(namespace) + shlex.quote(command)
 
     # Pass the close_fds argument to avoid the pidfile lock being held by
     # child processes
-    p = subprocess.Popen(command,
-                         shell=True,
+    p = subprocess.Popen(shlex.split(command),
```

Comment:
Pretty sure this doesn't work properly, if `namespace` is set and `command` has spaces in it.  What ends up getting run is `ip netns exec siganling "echo hello"` which will fail with `"echo hello": no such command`.

T'would be better to pass all commands around as arrays all the time rather than bouncing between strings and arrays.